astropy/nddata/utils.py

```diff
@@ -25,6 +28,21 @@
     "Cutout2D",
 ]
 
+if TYPE_CHECKING:
+    from numbers import Number
+    from typing import Literal, TypeAlias, TypeVar
+
+    from numpy.typing import NDArray
+
+    # These could be generalized as
+    # T = TypeVar("T")
+    # SingleOrTuple: TypeAlias = T | tuple[T, ...]
+    # but sphinx chokes on this definition at the time of writing
+    SingleOrTupleInt: TypeAlias = int | tuple[int, ...]
+    SingleOrTupleFloat: TypeAlias = float | tuple[float, ...]
```

Comment:
As far as Sphinx is concerned defining `SingleOrTupleInt` and `SingleOrTupleFloat` is not any better than `SingleOrTuple` because these definitions are still hidden from Sphinx in an `if TYPE_CHECKING:` block. You need to write out `int | tuple[int, ...]` and `float | tuple[float, ...]` in the type signatures.

Reply:
or maybe I can move them out of the `if TYPE_CHECKING` block instead ? (maybe even bring back my `SingleOrTuple` generic ??)

Reply:
I think you'd have to make `SingleOrTuple` a part of the public API or else Sphinx cannot link to it. I don't think it's worth it. `int | tuple[int, ...]` and `float | tuple[float, ...]` are a simple enough that we don't need aliases for them.